homeassistant/components/enocean/climate.py

```diff
@@ -0,0 +1,431 @@
+"""Support for Enocean Valve with integrated PID Controller."""
+
+import logging
+
+from enocean.protocol.constants import PACKET, RORG
+from enocean.protocol.packet import RadioPacket
+from simple_pid import PID
+import voluptuous as vol
+
+from homeassistant.components.climate import (
+    ATTR_TEMPERATURE,
+    PLATFORM_SCHEMA,
+    ClimateEntity,
+)
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_OFF,
+    DOMAIN,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_OFF,
+    PRESET_ECO,
+    PRESET_HOME,
+    PRESET_NONE,
+    SUPPORT_PRESET_MODE,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_BATTERY_LEVEL, CONF_ID, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .device import EnOceanEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "EnOcean Valve"
+
+SUPPORT_FLAGS = SUPPORT_TARGET_TEMPERATURE | SUPPORT_PRESET_MODE
+
+CONF_TARGET_TEMP = "target_temp"
+CONF_SETPOINTSELECTION = "setpointselection"
+CONF_PID_PARAMETER = "pid_parameter"
+
+DEFAULT_PID_PARAMETER_KP = 0.5
+DEFAULT_PID_PARAMETER_KI = 0.2
+DEFAULT_PID_PARAMETER_KD = 0
+
+SCHEMA_PID_PARAMETER = vol.Schema({"Kp": float, "Ki": float, "Kd": float})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
We don't allow new platforms with yaml config sections in this integration. This integration is under migration to use config flow. The existing platform yaml has not yet been migrated.

See https://github.com/home-assistant/core/pull/35927